language/move-analyzer/src/symbols.rs

```diff
@@ -2206,6 +2210,288 @@ pub fn on_hover_request(context: &Context, request: &Request, symbols: &Symbols)
     );
 }
 
+/// Returns (index of the function, text info of the function name)
+///
+/// The text info result includes the name and the range of the function in a given line.
+///
+/// A non-zero parenthesis_delta value means the current index is in a nested function scope.
```

Comment:
Please move the comments concerning algorithmic details for all functions (which are very useful, by the way!) to the beginning of the function itself, leaving the documentation comments more succinct.